### PR TITLE
[posix] install high priority kernel routes for OMR prefixes

### PR DESCRIPTION
**Pull request by simonlingoogle**
This commit allows a POSIX BR to install high priority kernel routes for OMR prefixes in Network Data.

The commit also sligitly revise the logs for ExternRoutes feature.

### CODE REVIEW

File: src/posix/platform/netif.cpp [hunk 1]
```diff
@@ -638,6 +647,131 @@ static otError DeleteExternalRoute(const otIp6Prefix &aPrefix)
...
+        else
+        {
+            sAddedOmrRoutes[i] = sAddedOmrRoutes[sAddedOmrRoutesNum - 1];
+            --sAddedOmrRoutesNum;
+            --i;
+            otLogInfoPlat("[netif] Successfully deleted an OMR route %s in kernel", prefixString);
+        }
+    }
+
+    while (otNetDataGetNextOnMeshPrefix(aInstance, &iterator, &config) == OT_ERROR_NONE)
```
Comment (superwhd) [inline src/posix/platform/netif.cpp:735]: To be consistent with the above section, may add a comment describing what this section does.

File: include/openthread/border_router.h [hunk 2]
```diff
@@ -117,6 +117,15 @@ otError otBorderRoutingGetOmrPrefix(otInstance *aInstance, otIp6Prefix *aPrefix)
  */
 otError otBorderRoutingGetOnLinkPrefix(otInstance *aInstance, otIp6Prefix *aPrefix);
 
+/**
+ * This method checks if the On Mesh Prefix is a valid off-mesh-routable (OMR) Prefix.
+ *
+ * @param[in]  aConfig    A pointer the the On Mesh Prefix inforamtion.
```
Comment (superwhd) [inline include/openthread/border_router.h:123]:
```suggestion
 * @param[in]  aConfig    A pointer to the On Mesh Prefix information.
```

typo

**Review by superwhd — APPROVED** [review]
LGTM with nits :+1:

**Comment by simonlingoogle** [comment]
close in favor of https://github.com/openthread/openthread/pull/7605

**Pull request closed without merging**